src/speed_challenge_ctrl.cpp

```diff
@@ -133,71 +120,105 @@ class SpeedChallenge{
 
-            point2_direction_vec_.x = starting_pos_2_prop_.x * point2_distance/prop_distance;
-            point2_direction_vec_.y = starting_pos_2_prop_.y * point2_distance/prop_distance;
+                // Direction vector from start to point 2 
 
-            //Direction vector from start to point 3 (Rotate 30 degrees counter clockwise)
+                point2_distance = prop_distance + avoidance_distance_p;
 
-            point3_direction_vec_.x = starting_pos_2_prop_.x * cos(avoidance_angle_p * M_PI/180) - starting_pos_2_prop_.x * sin(avoidance_angle_p * M_PI/180);
-            point3_direction_vec_.y = starting_pos_2_prop_.y * sin(avoidance_angle_p * M_PI/180) + starting_pos_2_prop_.y * cos(avoidance_angle_p * M_PI/180);
+                point2_direction_vec_.x = starting_pos_2_prop_.x * point2_distance/prop_distance;
+                point2_direction_vec_.y = starting_pos_2_prop_.y * point2_distance/prop_distance;
 
-            status = states::MOVE_TO_POINT1;
-            ROS_INFO("Moving to Point 1");
+                //Direction vector from start to point 3 (Rotate 30 degrees counter clockwise)
+
+                point3_direction_vec_.x = starting_pos_2_prop_.x * cos(avoidance_angle_p * M_PI/180) - starting_pos_2_prop_.y * sin(avoidance_angle_p * M_PI/180);
+                point3_direction_vec_.y = starting_pos_2_prop_.x * sin(avoidance_angle_p * M_PI/180) + starting_pos_2_prop_.y * cos(avoidance_angle_p * M_PI/180);
+
+                status = states::MOVE_TO_GATE; 
+
+            }
             break;
 
+         case states::MOVE_TO_GATE:
 
-        case states::MOVE_TO_POINT1:
+             ROS_INFO("Moving to Gate");
 
-            // Get point 1 
+             task_goal_pos_.point.x = gate_mid_point_.x;
+             task_goal_pos_.point.y = gate_mid_point_.y;
 
-            task_goal_pos_.point.x = point1_direction_vec_.x + starting_pos_.pose.position.y;
-            task_goal_pos_.point.y = point1_direction_vec_.y + starting_pos_.pose.position.x * -1;
-            task_goal_pos_.task.current_task = task_master::Task::SPEED_RUN;
+             task_goal_pos_.task.current_task = task_master::Task::SPEED_RUN;
             
-            ROS_INFO("Moving to Point 1");
-            task_pos_.publish(task_goal_pos_);
+             task_pos_.publish(task_goal_pos_);
 
-            if(isReached()){
+             if(isReached()){
 
-                status = states::MOVE_TO_POINT2;
+                 status = states::MOVE_TO_POINT1;
 
-                ROS_INFO("Point 1 reached");
-                ROS_INFO("Moving to Point 2");
-            }
-            break;
+                 ROS_INFO("Gate reached");
+                 ROS_INFO("Moving to Point 1");
+             }
+             break;
 
-        case states::MOVE_TO_POINT2:
+         case states::MOVE_TO_POINT1:
+
+         // Get point 1 
+
+             task_goal_pos_.point.x = point1_direction_vec_.x + starting_pos_.pose.position.x;
```

Comment:
I'm sure the math is all good I'm just wondering why the starting position needs to get added again here to point1? Also, Would it make more sense to add it when you instantiate point1, point2, .. and do all the math in one place, and then can set task_goal_pos to a single variable? Let me know if that doesn't make any sense cause my brain is kinda soup rn.